llvm/lib/Target/X86/AsmParser/X86AsmParser.cpp

```diff
@@ -3670,11 +3674,11 @@ bool X86AsmParser::ParseInstruction(ParseInstructionInfo &Info, StringRef Name,
 }
 
 bool X86AsmParser::processInstruction(MCInst &Inst, const OperandVector &Ops) {
-  if (ForcedVEXEncoding != VEXEncoding_VEX3 &&
+  if (AsmOptimize && ForcedVEXEncoding != VEXEncoding_VEX3 &&
       X86::optimizeInstFromVEX3ToVEX2(Inst, MII.get(Inst.getOpcode())))
     return true;
 
-  if (X86::optimizeShiftRotateWithImmediateOne(Inst))
+  if (AsmOptimize && X86::optimizeShiftRotateWithImmediateOne(Inst))
```

Comment:
@kongy does your assembler not optimize shifts/rotates by 1 to use the short form? That seems like a pretty basic optimization that has been around for x86 for a very long time.

Reply:
After some investigation, our assembler (Android ART) does implement the optimizeShiftRotateWithImmediateOne optimization. I guess we only need to turn off the VEX3ToVEX2 optimization.

Reply:
Do you still need this patch? If so, I guess you might need to change it to "x86-asm-vex3-to-vex2"?

Reply:
We ended up implementing the same optimization in our assembler. This change is no longer required.